web/src/integTest/java/com/microsoft/applicationinsights/azuretest/CompareTelemetryWithAzureExportTests.java

```diff
@@ -0,0 +1,342 @@
+
+    /**
+     * Clears the Azure queue from all existing messages
+     * @throws Exception
+     */
+    private void clearQueue() throws Exception {
+        CloudStorageAccount account = CloudStorageAccount.parse(config.getStorageConnectionString());
+        CloudQueueClient queueClient = account.createCloudQueueClient();
+        CloudQueue queue = queueClient.getQueueReference(config.getQueueName());
+
+        queue.downloadAttributes();
+        System.out.println("Before clearing: Queue contains " + queue.getApproximateMessageCount() + " items");
+        System.out.println("Clearing queue ...");
+        queue.clear();
+    }
+
+    private ArrayList<TelemetryItem> sentHttpGetRequests() throws Exception {
+
+        String serverAddress = config.getTestServerAddress();
+        int tomcat7Port = config.getTomcat7Port();
+        String applicationFollder = config.getApplicationFolder();
+
+        String requestId1 = UUID.randomUUID().toString();
+        String requestId2 = UUID.randomUUID().toString();
+        String requestId3 = UUID.randomUUID().toString();
+
+        URI uri1 = new URI("http://" + serverAddress + ":" + tomcat7Port + "/" + applicationFollder + "/books?id=Thriller&runId=" + runId + "&requestId=" + requestId1);
+        URI uri2 = new URI("http://" + serverAddress + ":" + tomcat7Port + "/" + applicationFollder + "/loan?title=Gone%20Girl&id=030758836x&subject=Thriller&runId=" + runId + "&requestId=" + requestId2);
+        URI uri3 = new URI("http://" + serverAddress + ":" + tomcat7Port + "/" + applicationFollder + "/notExistingWebFage?runId=" + runId + "&requestId=" + requestId3);
+
+        TelemetryItem expectedTelemetry1 = sendGet(uri1, requestId1);
+        TelemetryItem expectedTelemetry2 = sendGet(uri2, requestId2);
+        TelemetryItem expectedTelemetry3 = sendGet(uri3, requestId3);
+
+        ArrayList<TelemetryItem> expectedTelemetries = new ArrayList<TelemetryItem>();
+        expectedTelemetries.add(expectedTelemetry1);
+        expectedTelemetries.add(expectedTelemetry2);
+        expectedTelemetries.add(expectedTelemetry3);
+
+        return expectedTelemetries;
+    }
+
+    /**
+     * Sends HTTP GET request and returns the expected telemetry from AppInsights
+     * @param uri The URI for the request
+     * @return The expected telemetry items from AppInsights
+     * @throws Exception
+     */
+    private TelemetryItem sendGet(URI uri, String requestId) throws Exception {
+        HttpURLConnection connection = (HttpURLConnection) uri.toURL().openConnection();
+
+        // optional default is GET
+        connection.setRequestMethod("GET");
+
+        //add request header
+        connection.setRequestProperty("User-Agent", config.getUserAgent());
+
+        System.out.println("Sending 'GET' request to URL: " + uri.toString());
+        Integer responseCode = connection.getResponseCode();
+        System.out.println("Response Code : " + responseCode);
+
+        // Create expected result
```

Comment:
Please refactor by separating the logic for composing and sending the GET request and calculating the expected telemetry from the URL.